사이드바 메뉴 컴포넌트 제작

jinhyukoo opened the pull request:
## 📕 제목

#14 

![SideBarMenu](https://user-images.githubusercontent.com/46101366/98441544-6b7f6000-2142-11eb-9137-814e41a84c1f.gif)

## 📗 작업 내용

> 구현 내용 및 작업 했던 내역

- [x] 사이드바 메뉴 컴포넌트 UI 제작
- [x] 마우스 오버시 이미지의 색상도 변하게 설정
- [x] 컴포넌트에 props로 함수 전달하면 Header 부분 클릭시 그 함수 실행

## 📘 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- file-loader 웹팩 설정한 부분을 같이 올리려 했는데 우선 우진님께서 필요 없다고 하셔서 포함시키지 않았습니다. 다른 분들도 이미지 사용하실 때 문제가 생기시면 설정한 부분 추가로 올리겠습니다.

jinhyukoo committed:
feat: [FE] SideBarMenu 컴포넌트 UI 제작

jinhyukoo committed:
feat: [FE] SideBarMenuHeader에 hover 효과 추가

jinhyukoo committed:
chore: [FE] index.js에 SideBarMenu 컴포넌트 경로 추가

jinhyukoo committed:
feat: [FE] Header 부분 클릭시 이벤트 발생

props로 함수를 전달해주면 Header 클릭시 특정 이벤트를 발생시킬 수 있게 설정

jinhyukoo committed:
chore: [FE] color.js에 색상 추가 및 파일에 적용

jinhyukoo committed:
chore: [FE] 웹팩 설정 추가

이미지 파일을 불러오기 위해 file-loader를 웹팩 설정에 추가함

wooojini commented on frontend/src/components/SidebarMenu/SidebarMenu.jsx:
```diff
@@ -0,0 +1,56 @@
+                    <img src={settingIcon} alt="setting" width="16" height="16" />
+                </div>
+            </StyledSideBarMenuHeader>
+            <StyledSideBarMenuBody>{children}</StyledSideBarMenuBody>
```
children을 활용해서 메뉴 내부에 원하는 컴포넌트를 끼워넣을 수 있도록 해주셨군요 :-)

wooojini commented on frontend/src/components/SidebarMenu/SidebarMenu.jsx:
```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import styled from "styled-components";
+import { color } from "@style/color";
+import settingIcon from "../../../public/images/setting-icon.png";
```
webpack.config.common.js 파일에 alias 설정이 되어있습니다! 이미지경로를 public/imgs로 변경하고, alias를 사용해주세요 :-)
```
           alias: {
                "@config": path.resolve(__dirname, "config"),
                "@components": path.resolve(__dirname, "src/components"),
                "@pages": path.resolve(__dirname, "src/pages"),
                "@style": path.resolve(__dirname, "src/common/style"),
                "@utils": path.resolve(__dirname, "src/common/utils"),
                "@imgs": path.resolve(__dirname, "public/imgs"),
                "@hook": path.resolve(__dirname, "src/common/hook"),
                "@context": path.resolve(__dirname, "src/common/context")
            }
```

Do-ho replied:
ㅠㅠ 제가 빠르게 이미지를 따드리겠습니다.

jinhyukoo replied:
혹시 이미지는 alias로 가져오려면 어떤식으로 가져와야 할까요? export를 시키지 않아서 그런지 그냥 경로로 접근을 하면 가져와지질 않네요...!

wooojini replied:
index.js를 만들고 export해서 경로로 가져와도 좋고, 그냥 파일만 위치시키고, @imgs/....과 같이 가져와도 좋아보이네요!

wooojini commented on frontend/webpack.config.common.js:
```diff
@@ -66,6 +66,14 @@ module.exports = (env) => {
+                            loader: "file-loader"
+                        }
+                    ]
                 }
```
앗! 지금보니 파일을 로드하는 설정이 아예 빠져있었군요! 혹시 다음과 같이 변경해 볼 수 있을까요?? url-loader 설정에서 limit 옵션 설정을 해주면, 설정한 용량 이상의 이미지는 자동으로 file-loader를 사용합니다. 아래설정은 10KB 미만의 파일은 url-loader를 그 이상의 파일은 file-loader를 사용하는 설정입니다!
```
               {
                    test: /\.(png|jpe?g|gif)$/i,
                    use: {
                        loader: "url-loader",
                        options: {
                            publicPath: "/",
                            name: "[name].[ext]?[hash]",
                            limit: 10000,
                        },
                    }
                }
```

Do-ho replied:
[해당 글](https://jeonghwan-kim.github.io/js/2017/05/22/webpack-file-loader.html)을 보니 우진님이 말한 내용의 설정이 좋아보입니다!

jinhyukoo replied:
해당 부분 수정했습니다!

Do-ho reviewed (approved):
사이드바 메뉴 작성하느라 고생 많으셨습니다 :)
빠르게.. 이미지를 따서 드리도록 할게요!

youngxpepp reviewed (approved):
css 한 수 배워갑니다!
잘하시는 분들 컴포넌트 만드는걸 많이 참조해야겠네요. 승인하겠습니다!

jinhyukoo committed:
chore: [FE] 웹팩 설정 변경

- url-loader와 file-loader를 필요한 상황에 따라 사용
- 10KB 미만일 때는 url-loader, 그 외엔 file-loader 사용

wooojini reviewed (approved):
고생하셨습니다!!! Approve 할게요~

jinhyukoo committed:
Merge branch 'develop' into feature/14-sidebarmenu
